EchoPro/survey.py

```diff
@@ -36,741 +23,440 @@ class Survey:
+        data using parameters obtained from the configuration
+        files. This will generate data attributes associated with the tags
+        defined in both the configuration yml files and the reference CONFIG_MAP
+        and LAYER_NAME_MAP dictionaries.
+        """
+
+        ### Check whether data files defined from the configuration file exists
+        # Generate flat JSON table comprising all configuration parameter names
+        flat_configuration_table = pd.json_normalize(self.config).filter(regex="filename")
+
+        # Parse the flattened configuration table to identify data file names and paths
+        parsed_filenames = flat_configuration_table.values.flatten()
+
+        # Append the root directory to the parsed data file names
+        data_paths = [Path(self.config['data_root_dir'] + '/' + file) for file in parsed_filenames]
+
+        # Evaluate whether either file is missing
+        data_existence = [path.exists() for path in data_paths]
+
+        # Assign the existence status to each configuration file for error evaluation
+        # Error evaluation and print message (if applicable)
+        if not all(data_existence):
+            missing_data = [ files for files, exists in zip( parsed_filenames, data_existence ) if not exists ]
+            raise FileNotFoundError(f"The following data files do not exist: {missing_data}")
+        
+        # Initialize data attributes ! 
+        setattr( self , 'acoustics' , { } )
+        setattr( self , 'biology' , { } )
+        setattr( self , 'spatial' , { } )
+        setattr( self , 'statistics' , { } )
+
+        ### Data validation and import
+        # Iterate through each file defined/detected in 'data_paths' and 'parsed_filenames'
+
+        for file , path in zip(parsed_filenames, data_paths):
+
+            # Parse config to find appropriate settings
+            config_name = flat_configuration_table.apply(lambda row: row[row == file].index, axis=1)
+            
+            # Convert to string
+            config_identifier = config_name[0].tolist()[0].lower()
+
+            # Extract appropriate dataset name from the file path
+            # -- Based on tail-end of file path string to avoid potential 
+            # -- directory name issues
+            dataset = [x for x in [*CONFIG_MAP.keys()] if x.lower() in config_identifier][0]
+            
+            # Determine data layer name
+            datalayer = [x for x in [*CONFIG_MAP.get(dataset)] if x in config_identifier.split('.')][0]
+
+            # Define the data layer name 
+            # -- Based on the lattermost portion of the file path string
+            # Create list for parsing the hard-coded API dictionary
+            if dataset == 'biological':                
+                region_id = [x for x in ['US', 'CAN'] if x.lower() in config_identifier][0]
+                sheet_name = self.config.get(dataset).get(datalayer).get(region_id).get('sheetname')
```

Comment:
This makes the code more readable because it is easy to directly see the multi-layer dict structure:
```suggestion
                sheet_name = self.config[dataset][datalayer][region_id]['sheetname']
```